docs/maintaining/testing.rst

```diff
@@ -0,0 +1,57 @@
+Testing
+=======
+
+When making a contribution to opsdroid, it's important that tests are written to validate that the functionality works, and continues to work in the future.
+
+opsdroid is currently slowly migrating from `unittest <https://docs.python.org/3.8/library/unittest.html>`__ to `pytest <https://docs.pytest.org>`__ as it's testing framework.
```

Comment:
Can we link to the issue here and also add todo item to the issue to remove this when done.

Reply:
done.